File: src/test/java/com/guardtime/ksi/samples/SignatureContentSamples.java
package com.guardtime.ksi.samples;

import java.io.IOException;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import com.guardtime.ksi.KSI;
import com.guardtime.ksi.exceptions.KSIException;
import com.guardtime.ksi.publication.PublicationData;
import com.guardtime.ksi.unisignature.AggregationHashChain;
import com.guardtime.ksi.unisignature.CalendarAuthenticationRecord;
import com.guardtime.ksi.unisignature.CalendarHashChain;
import com.guardtime.ksi.unisignature.KSISignature;
import com.guardtime.ksi.unisignature.SignatureData;
import com.guardtime.ksi.unisignature.SignaturePublicationRecord;

public class SignatureContentSamples extends KsiSamples {

    @Before
    public void setUp() throws KSIException {
        setUpKsi();
    }

    @After
    public void tearDown() {
        tearDownKsi();
    }

    /**
     * Prints information found in the given signature's publication record.
     */
    @Test
    public void printPublicationRecord() throws IOException, KSIException {
        KSI ksi = getKsi();

        KSISignature signature = ksi.read(getFile("signme.txt.extended-ksig"));

        SignaturePublicationRecord spr = signature.getPublicationRecord();
        if (spr != null)
            System.out.println("printPublicationRecord > publication time >" + spr.getPublicationTime());
        else
            System.out.println("printPublicationRecord > No publication record in signature");
    }

    /**
     * Prints information found in the given signature's aggregation hash chain.
     */
    @Test
    public void printAhc() throws IOException, KSIException {
        KSI ksi = getKsi();
        KSISignature signature = ksi.read(getFile("signme.txt.extended-ksig"));

        for (AggregationHashChain ahc : signature.getAggregationHashChains()) {
            System.out.println("printAhc > link count > " + ahc.getChainLinks().size());
        }
    }

    /**
     * Prints information found in the given signature's calendar hash chain.
     */
    @Test
    public void printChc() throws IOException, KSIException {
        KSI ksi = getKsi();

        KSISignature signature = ksi.read(getFile("signme.txt.extended-ksig"));

        CalendarHashChain chc = signature.getCalendarHashChain();
        System.out.println("printChc > publication time > " + chc.getPublicationTime());
        System.out.println("printChc > registration time > " + chc.getRegistrationTime());

    }

    /**
     * Prints the identity in the signature.
     */
    @Test
    public void printIdentity() throws IOException, KSIException {
        KSI ksi = getKsi();
        KSISignature signature = ksi.read(getFile("signme.txt.extended-ksig"));

        System.out.println("printIdentity > " + signature.getIdentity());
    }

    /**
     * Prints the signing (aggregation) time of the signature.
     */
    @Test
    public void printSigningTime() throws IOException, KSIException {
        KSI ksi = getKsi();

        KSISignature signature = ksi.read(getFile("signme.txt.extended-ksig"));

        System.out.println("printSigningTime > " + signature.getAggregationTime());
    }

    /**
     * Prints the RSA signature type of the calendar authentication record.
     */
    @Test
    public void printCalendarAuthenticationRecord() throws IOException, KSIException {
        KSI ksi = getKsi();

        KSISignature signature = ksi.read(getFile("signme.txt.extended-ksig"));

        CalendarAuthenticationRecord car = signature.getCalendarAuthenticationRecord();
        PublicationData publicationData = car.getPublicationData();

        SignatureData signatureData = car.getSignatureData();
        publicationData.getPublicationString();

        System.out.println("printCalendarAuthenticationRecord > signature type > " + signatureData.getSignatureType());
    }
}

File: ksi-net-sdk-samples/ksi-net-sdk-samples/SignatureContentSamples.cs
using System;
using Guardtime.KSI.Signature;
using Guardtime.KSI.Utils;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Guardtime.Ksi.Samples
{
    [TestClass]
    public class SignatureContentSamples : KsiSamples
    {
        /// <summary>
        /// Prints information found in the given signature's publication record.
        /// </summary>
        [TestMethod]
        public void PrintPublicationRecord()
        {
            IKsiSignature signature = LoadExtendedSignature();

            if (signature.PublicationRecord != null)
            {
                Console.WriteLine("PrintPublicationRecord > publication time >" + Util.ConvertUnixTimeToDateTime(signature.PublicationRecord.PublicationData.PublicationTime));
            }
            else
            {
                Console.WriteLine("PrintPublicationRecord > No publication record in signature");
            }
        }

        /// <summary>
        /// Prints information found in the given signature's aggregation hash chain.
        /// </summary>
        [TestMethod]
        public void PrintAggregationHashChain()
        {
            IKsiSignature signature = LoadExtendedSignature();

            foreach (AggregationHashChain ahc in signature.GetAggregationHashChains())
            {
                Console.WriteLine("PrintAggregationHashChain > chain identity > " + ahc.GetChainIdentity());
            }
        }

        /// <summary>
        /// Prints information found in the given signature's calendar hash chain.
        /// </summary>
        [TestMethod]
        public void PrintCalendarHashChain()
        {
            IKsiSignature signature = LoadExtendedSignature();

            Console.WriteLine("PrintCalendarHashChain > publication time > " + Util.ConvertUnixTimeToDateTime(signature.CalendarHashChain.PublicationTime));
            Console.WriteLine("PrintCalendarHashChain > registration time > " + Util.ConvertUnixTimeToDateTime(signature.CalendarHashChain.RegistrationTime));
        }

        /// <summary>
        /// Prints the identity in the signature.
        /// </summary>
        [TestMethod]
        public void PrintIdentity()
        {
            IKsiSignature signature = LoadExtendedSignature();

            Console.WriteLine("PrintIdentity > " + signature.Identity);
        }

        /// <summary>
        /// Prints the signing (aggregation) time of the signature.
        /// </summary>
        [TestMethod]
        public void PrintSigningTime()
        {
            IKsiSignature signature = LoadExtendedSignature();

            Console.WriteLine("PrintSigningTime > " + Util.ConvertUnixTimeToDateTime(signature.AggregationTime));
        }

        /// <summary>
        /// Prints the RSA signature type of the calendar authentication record.
        /// </summary>
        [TestMethod]
        public void PrintCalendarAuthenticationRecord()
        {
            IKsiSignature signature = LoadUnextendedSignature();

            Console.WriteLine("PrintCalendarAuthenticationRecord > signature type > " + signature.CalendarAuthenticationRecord.SignatureData.SignatureType);
        }
    }
}